source/tyr/tyr/resources.py

```diff
@@ -138,6 +138,39 @@ def post(self, instance_name):
 
         return {'message': 'OK'}, 200
 
+    def delete(self, id=None, instance_name=None):
+        parser = reqparse.RequestParser()
+        parser.add_argument('state', type=str, required=False, case_sensitive=True, help='filter by job state')
+        parser.add_argument('confirm', type=str, required=True, case_sensitive=True, help='confirm the deletion')
+        args = parser.parse_args()
+
+        if args["confirm"] == "yes":
+            if id:
+                jobs_to_delete = [models.Job.get(id)]
+            else:
+
+                query = models.Job.query
+                if "state" in args and args["state"]:
+                    query = query.filter_by(state=args["state"])
+                if instance_name:
+                    query = query.join(models.Instance).filter(models.Instance.name == instance_name)
+                jobs_to_delete = query.all()
+
+            if not jobs_to_delete:
+                return json.loads('{}'), 204
+
+            for job in jobs_to_delete:
+                try:
+                    db.session.delete(job)
+                except Exception:
+                    logging.exception("Failed to delete Job {} - Operation aborted".format(job.id))
+                    raise
+
+            db.session.commit()
+            return json.loads('{}'), 200
+        else:
+            return {'message': 'confirm the deletion'}, 400
```

Comment:
```suggestion
            return {'message': 'confirm the deletion with parameter "confirm=yes"'}, 400
```
Being explicit is better not to go back to the source code if you ever want to use the endpoint.